src/common/pcqueue.h

```diff
@@ -22,24 +22,58 @@
 
 #include "common/platform.h"
 
-#include <inttypes.h>
-
-inline void queue_deleter_dummy(uint8_t *) {
-}
+#include <condition_variable>
+#include <cstdint>
+#include <functional>
+#include <mutex>
+#include <queue>
 
 template<typename T>
-void queue_deleter_delete(uint8_t *p) {
+void deleterByType(uint8_t *p) {
 	delete ((T*)p);
 }
 
+inline void deleterDummy(uint8_t * /*unused*/) {}
+
+class PCQueue {
```

Comment:
Consider adding minimum the class description documentation. Also: when is it recomented to use it? Maybe how to use it?

Reply:
Done